pkg/apis/internal.acorn.io/v1/appinstance.go

```diff
@@ -55,24 +56,25 @@ const (
 )
 
 type AppInstanceSpec struct {
-	Labels              []ScopedLabel    `json:"labels,omitempty"`
-	Annotations         []ScopedLabel    `json:"annotations,omitempty"`
-	Image               string           `json:"image,omitempty"`
-	Stop                *bool            `json:"stop,omitempty"`
-	DevMode             *bool            `json:"devMode,omitempty"`
-	Profiles            []string         `json:"profiles,omitempty"`
-	Volumes             []VolumeBinding  `json:"volumes,omitempty"`
-	Secrets             []SecretBinding  `json:"secrets,omitempty"`
-	Environment         []NameValue      `json:"environment,omitempty"`
-	PublishMode         PublishMode      `json:"publishMode,omitempty"`
-	TargetNamespace     string           `json:"targetNamespace,omitempty"`
-	Links               []ServiceBinding `json:"services,omitempty"`
-	Ports               []PortBinding    `json:"ports,omitempty"`
-	DeployArgs          GenericMap       `json:"deployArgs,omitempty"`
-	Permissions         []Permissions    `json:"permissions,omitempty"`
-	AutoUpgrade         *bool            `json:"autoUpgrade,omitempty"`
-	NotifyUpgrade       *bool            `json:"notifyUpgrade,omitempty"`
-	AutoUpgradeInterval string           `json:"autoUpgradeInterval,omitempty"`
+	Labels              []ScopedLabel                          `json:"labels,omitempty"`
+	Annotations         []ScopedLabel                          `json:"annotations,omitempty"`
+	Image               string                                 `json:"image,omitempty"`
+	Stop                *bool                                  `json:"stop,omitempty"`
+	DevMode             *bool                                  `json:"devMode,omitempty"`
+	Profiles            []string                               `json:"profiles,omitempty"`
+	Volumes             []VolumeBinding                        `json:"volumes,omitempty"`
+	Secrets             []SecretBinding                        `json:"secrets,omitempty"`
+	Environment         []NameValue                            `json:"environment,omitempty"`
+	PublishMode         PublishMode                            `json:"publishMode,omitempty"`
+	TargetNamespace     string                                 `json:"targetNamespace,omitempty"`
+	Links               []ServiceBinding                       `json:"services,omitempty"`
+	Ports               []PortBinding                          `json:"ports,omitempty"`
+	DeployArgs          GenericMap                             `json:"deployArgs,omitempty"`
+	Permissions         []Permissions                          `json:"permissions,omitempty"`
+	AutoUpgrade         *bool                                  `json:"autoUpgrade,omitempty"`
+	NotifyUpgrade       *bool                                  `json:"notifyUpgrade,omitempty"`
+	AutoUpgradeInterval string                                 `json:"autoUpgradeInterval,omitempty"`
+	Memory              map[string]corev1.ResourceRequirements `json:"memory,omitempty"`
```

Comment:
In our design discussion, we said that this should just be a simple int, not a the core type.

So, the conversion to a resource requirement shouldnt happen until you're in the controller and translating AppInstance to a deployment.